dev/webpack.config.js

```diff
@@ -0,0 +1,161 @@
+/* eslint import/no-commonjs: 0 */
+const path = require('path');
+const without = require('lodash/without');
+
+const webpack = require('webpack');
+const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const HtmlWebpackPluging = require('html-webpack-plugin');
+const BabiliPlugin = require('babili-webpack-plugin');
+const AutoDllPlugin = require('autodll-webpack-plugin');
+const HappyPack = require('happypack');
+
+const __DEV__ = process.env.NODE_ENV === 'development';
+const clean = arr => arr.filter(item => item !== false);
+
+module.exports = {
```

Comment:
I was using different configurations before and merging them, but it was too complicated at some point. 

When you have different depth level of objects and arrays it's just a mess to have something like that to maintain. You are lacking of context, must have multiple files open and since this is only configuration I don't think we need to separate that much the concerns as we do in our codebase.

For instance overriding some options and adding new ones from a common configuration:

prod.webpack.config.js
```js
const commonConfig = require('./common.config.webpack.js');
const prodConfig = Object.assign({}, commonConfig);

prodConfig.entry = {
  bundle: './dev/app'
};

// change the js loader config
commonConfig.module.rules[3].use[1].options = {sourceMap: false};

// remove happypack
commonConfig.plugins.splice(3, 1);

// add minify
commonConfig.plugins.push(new BabiliPlugin());
```

I have also tried the "duplication" strategy like `create-react-app` does. It is also too complicated to maintain when you update the configuration you have the replicate all the changes in between configurations.

If we really want to separate configurations, we should look https://github.com/Fitbit/webpack-config